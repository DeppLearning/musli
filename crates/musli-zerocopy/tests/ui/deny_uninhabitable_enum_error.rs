use musli_zerocopy::ZeroCopy;

#[derive(Debug, Clone, Copy, PartialEq, ZeroCopy)]
#[repr(u8)]
enum Uninhabitable {}

fn main() {
}
