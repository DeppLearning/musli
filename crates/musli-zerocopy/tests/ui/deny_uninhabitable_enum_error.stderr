error[E0084]: unsupported representation for zero-variant enum
 --> tests/ui/deny_uninhabitable_enum_error.rs:4:1
  |
4 | #[repr(u8)]
  | ^^^^^^^^^^^
5 | enum Uninhabitable {}
  | ------------------ zero-variant enum

warning: unreachable expression
 --> tests/ui/deny_uninhabitable_enum_error.rs:3:41
  |
3 | #[derive(Debug, Clone, Copy, PartialEq, ZeroCopy)]
  |                                         ^^^^^^^^
  |                                         |
  |                                         unreachable expression
  |                                         any code following this `match` expression is unreachable, as all arms diverge
  |
  = note: `#[warn(unreachable_code)]` on by default
  = note: this warning originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)
